frontend/controller/actions/group.js

```diff
@@ -0,0 +1,89 @@
+import { RULE_THRESHOLD } from '@model/contracts/voting/rules.js'
+import { GIErrorUIRuntimeError } from '@model/errors.js'
+
+import imageUpload from '@utils/imageUpload.js'
+import { merge } from '@utils/giLodash.js'
+// import { CONTRACT_IS_SYNCING } from '@utils/events.js'
+import L from '@view-utils/translations.js'
+
+export default sbp('sbp/selectors/register', {
+  'gi.actions/group/create': async function ({
+    name,
+    picture,
+    sharedValues,
+    mincomeAmount,
+    mincomeCurrency,
+    thresholdChange,
+    thresholdMemberApproval,
+    thresholdMemberRemoval
+  }, cb) {
+    let finalPicture = `${window.location.origin}/assets/images/default-group-avatar.png`
+
+    if (picture) {
+      try {
+        finalPicture = await imageUpload(picture)
+      } catch (e) {
+        console.error('Failed to upload the group picture', e)
+        throw new GIErrorUIRuntimeError(L('Failed to upload the group picture. {codeError}', { codeError: e.message }))
+      }
+    }
+
+    try {
+      const initialInvite = createInvite({ quantity: 60, creator: INVITE_INITIAL_CREATOR })
+      const entry = sbp('gi.contracts/group/create', {
+        invites: {
+          [initialInvite.inviteSecret]: initialInvite
+        },
+        settings: {
+          // authorizations: [contracts.CanModifyAuths.dummyAuth()], // TODO: this
+          groupName: name,
+          groupPicture: finalPicture,
+          sharedValues,
+          mincomeAmount: +mincomeAmount, // ensure this is a number
+          mincomeCurrency: mincomeCurrency,
+          proposals: {
+            // TODO: make the UI support changing the rule type, so that we have
+            //       a component for RULE_DISAGREEMENT as well
+            [PROPOSAL_GROUP_SETTING_CHANGE]: merge({},
+              proposals[PROPOSAL_GROUP_SETTING_CHANGE].defaults,
+              { ruleSettings: { [RULE_THRESHOLD]: { threshold: thresholdChange } } }
+            ),
+            [PROPOSAL_INVITE_MEMBER]: merge({},
+              proposals[PROPOSAL_INVITE_MEMBER].defaults,
+              { ruleSettings: { [RULE_THRESHOLD]: { threshold: thresholdMemberApproval } } }
+            ),
+            [PROPOSAL_REMOVE_MEMBER]: merge({},
+              proposals[PROPOSAL_REMOVE_MEMBER].defaults,
+              { ruleSettings: { [RULE_THRESHOLD]: { threshold: thresholdMemberRemoval } } }
+            ),
+            [PROPOSAL_PROPOSAL_SETTING_CHANGE]: proposals[PROPOSAL_PROPOSAL_SETTING_CHANGE].defaults,
+            [PROPOSAL_GENERIC]: proposals[PROPOSAL_GENERIC].defaults
+          }
+        }
+      })
+      await sbp('backend/publishLogEntry', entry)
+      return entry.hash()
+    } catch (e) {
+      throw new GIErrorUIRuntimeError(L('Failed to create the group: {codeError}', { codeError: e.message }))
+    }
+  },
+  // REVIEW these 2...
+  'gi.actions/group/createAndSubscribe': async function () {},
+  'gi.actions/group/createAndSubscribeWait': async function () {},
```

Comment:
Same question as above. Why not just creating a method `group/create` that by default it subscribes and waits. And if we really have a valid reason/case for each method, at least we could pass those as options.

```
await sbp('gi.actions/group/create', groupData, { subscribe: true, wait: true })
```

Reply:
Yes, that's actually a much cleaner API! (with `subscribe` renamed to `sync` per the above comment)

I like it. 👍 

Only thing is, if, [per the above commentary](https://github.com/okTurtles/group-income-simple/pull/801/files#r368738345), you test and find out that the implementation of `sync` already results in a wait, then you don't need the `wait` parameter here.